internal/release/github.go

```diff
@@ -129,20 +130,42 @@ func getOwnerRepoFromReleaseURL(url string) (string, string, error) {
 	return owner, repo, nil
 }
 
-// LoadPluginReleases loads the plugin-releases.json file from the specified GitHub release.
+// DownloadPluginReleasesToDir loads the plugin-releases.json file from the specified GitHub release.
 // It will additionally verify the minisign signature of the release if passed a valid minisign.PublicKey.
-func (c *Client) LoadPluginReleases(ctx context.Context, release *github.RepositoryRelease, publicKey minisign.PublicKey) (*PluginReleases, error) {
-	releasesJSONBytes, err := c.downloadAsset(ctx, release, PluginReleasesFile)
+// It will download both the plugin-releases.json and plugin-releases.json.minisig to the specified dir.
+func (c *Client) DownloadPluginReleasesToDir(
+	ctx context.Context,
+	release *github.RepositoryRelease,
+	publicKey minisign.PublicKey,
+	dir string,
+) (*PluginReleases, error) {
+	releasesJSONBytes, releasesJSONLastModified, err := c.downloadAsset(ctx, release, PluginReleasesFile)
 	if err != nil {
 		return nil, err
 	}
-	releasesJSONMinisigBytes, err := c.downloadAsset(ctx, release, PluginReleasesSignatureFile)
+	releasesJSONMinisigBytes, releasesJSONMinisigLastModified, err := c.downloadAsset(ctx, release, PluginReleasesSignatureFile)
 	if err != nil && !errors.Is(err, ErrNotFound) {
 		return nil, err
 	}
 	if !publicKey.Equal(minisign.PublicKey{}) && !minisign.Verify(publicKey, releasesJSONBytes, releasesJSONMinisigBytes) {
 		return nil, fmt.Errorf("release %s file %q doesn't match signature %q", release.GetName(), PluginReleasesFile, PluginReleasesSignatureFile)
 	}
+	pluginReleasesDownload := filepath.Join(dir, PluginReleasesFile)
+	if err := os.WriteFile(pluginReleasesDownload, releasesJSONBytes, 0644); err != nil {
+		return nil, err
+	}
+	if err := os.Chtimes(pluginReleasesDownload, releasesJSONLastModified, releasesJSONLastModified); err != nil {
+		return nil, err
+	}
+	if releasesJSONMinisigBytes != nil {
+		pluginReleasesSignatureDownload := filepath.Join(dir, PluginReleasesSignatureFile)
+		if err := os.WriteFile(pluginReleasesSignatureDownload, releasesJSONMinisigBytes, 0644); err != nil {
+			return nil, err
+		}
+		if err := os.Chtimes(pluginReleasesSignatureDownload, releasesJSONMinisigLastModified, releasesJSONMinisigLastModified); err != nil {
```

Comment:
I don't quite understand why we care about the time component at all here, I think I'm missing something...

Reply:
Ohhh, we're looping over all the assets and then updating the file to whatever the latest release timestamp is. But why does this matter though? Shouldn't this be based on the release name which has a date component..

Reply:
I'm trying to keep times in sync with the times on the release - it lets me tell at a glance when things changed last.